x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/parser/LogicalPlanBuilder.java

```diff
@@ -506,7 +507,8 @@ public LogicalPlan visitMetricsCommand(EsqlBaseParser.MetricsCommandContext ctx)
             false,
             List.of(new MetadataAttribute(source, MetadataAttribute.TSID_FIELD, DataType.KEYWORD, false)),
             IndexMode.TIME_SERIES,
-            null
+            null,
+            "FROM TS"
```

Comment:
This depends on the fact that a `METRICS idx ...` query could result in different execution plans:
- `METRICS idx | ...` (no grouping/stats) -> `UnresolvedRelation` (same as `FROM idx`)
- `METRICS idx max(...) BY ....` -> `UnresolvedRelation + Aggregate`

In the first case, to know it's a METRICS and not a FROM, I just store the command name.
In the second case, the Aggregate contains all the information to know it's actually a METRICS command (`AggregateType.METRICS`), but I don't want the UnresolvedRelation to account for a FROM or for a METRICS.

> because the query uses metrics specific aggregations and the indices must be time series indices

At this stage it doesn't really matter if the query is correct or not (eg. if the index is not a time series). The query could fail, but I'll still record the "intention" (ie. the usage of METRICS command)

> I am curious on the reason for not using a METRICS feature in the second case as well? (I am not saying it's wrong; I may be missing an important piece of knowledge)

I could use "METRICS" here and something else in Aggregate (eg. "METRICS STATS"). I cannot use METRICS in both cases, otherwise I'll increment the counter twice. 
I don't have a real preference between the two choices, the "FROM TS" is someway irrelevant for our statistics, and so "METRICS STATS" would be; what we really want is to record exactly one "METRICS"

> There is also one other implication here and I think it will affect how telemetry is reported. There are ongoing discussions about from time_series metadata _tsid, @timestamp | where index_mode == "time_series" | sort _tsid, @timestamp being equivalent with metrics time_series. And here the first example would also be a "METRICS" telemetry data.

Good point, we'll have to keep an eye on it.
Anyway, what we are tracking here is the syntax used, so if the user writes a query like `from time_series metadata _tsid, @timestamp | where index_mode == "time_series" | sort _tsid, @timestamp`, I think it's correct to report it as one FROM, one WHERE and one SORT, while if they write the query like `METRICS time_series` it would be correct to report it as one METRICS only.